src/OpenSage.Game/Logic/Player.cs

```diff
@@ -883,6 +883,11 @@ public GameObject FindClosestSupplyWarehouse(GameObject supplyGatherer)
             {
                 var supplyWarehouse = _game.GameLogic.GetObjectById(supplyWarehouseId);
 
+                if (supplyWarehouse == null)
```

Comment:
Unrelated to animation scaling - I just hit a crash here when loading a save file that had supply warehouses in it.